src/twisted/names/common.py

```diff
@@ -153,13 +153,21 @@ def lookupAllRecords(self, name, timeout=None):
     def getHostByName(self, name, timeout=None, effort=10):
         name = dns.domainString(name)
         # XXX - respect timeout
-        # XXX - this should do A and AAAA lookups, not ANY (see RFC 8482).
-        # https://twistedmatrix.com/trac/ticket/9691
-        d = self.lookupAllRecords(name, timeout)
-        d.addCallback(self._cbRecords, name, effort)
+        d = self.lookupAddress(name, timeout)
+        d.addCallback(self._cbRecordsA, name, effort, timeout)
         return d
 
-    def _cbRecords(self, records, name, effort):
+    def _cbRecordsA(self, records, name, effort, timeout):
+        (ans, auth, add) = records
+        result = extractRecord(self, dns.Name(name), ans + auth + add, effort)
+        if not result:
+            # if A lookup doesn't provide IP then lookup AAAA
```

Comment:
I think it should be wise to lookup AAAA records regardless of A results, it makes IPv6 stand on even field as IPv4, as perceived DNS requests (A vs AAAA) would even out, and it could help further adoption of IPv6

Reply:
This seems like it might make a good follow-up.

Reply:
What do you mean by that? Make another PR which applies this suggestion?

Reply:
File a ticket that describes the problem or the improved desired behavior and then make a PR that resolves it, yes.

Reply:
I agree with @ShadowJonathan that it should make queries for both A and AAAA by default, probably in parallel. It might also be worth making this behavior configurable for hosts that are ipv4-only, That seems like a good thing to put into a separate ticket+PR unless @taroved feels the need to do it in this one.

Reply:
@wiml I don't know how to do it in parallel. I think that it good thing to put into a separate ticket+PR